packages/next-auth/src/lib/index.ts

```diff
@@ -59,6 +59,33 @@ export interface NextAuthConfig extends Omit<AuthConfig, "raw"> {
   }
 }
 
+function mergeCookiesAndSetCookiesHeaders(headers: Headers): string {
+  const setCookies = headers.get("set-cookie")
+  const cookies = headers.get("cookie");
+
+  if(!setCookies || (!cookies && !setCookies)) {
+    return cookies ?? "";
+  }
+
+  // We need to reparse the headers because the headers from Next returns [] on getSetCookie().
+  const headersToParse = new Headers(headers);
+  const setCookiesHeaders = headersToParse.getSetCookie();
+  const cookiesArray = cookies?.split(";") ?? [];
```

Comment:
This won't be reliable, since dates can contain the `;` character, which will produce an  incorrect cookie array.